curriculum/test/test-challenges.js

```diff
@@ -86,59 +87,115 @@ spinner.text = 'Populate tests.';
 let browser;
 let page;
 
-runTests();
+setup()
+  .then(runTests)
+  .catch(err => {
+    cleanup();
+    // setting the error code because node does not (yet) exit with a non-zero
+    // code on unhandled exceptions.
+    process.exitCode = 1;
+    throw err;
+  });
+
+async function setup() {
+  // liveServer starts synchronously
+  liveServer.start({
+    host: '127.0.0.1',
+    port: '8080',
+    root: path.resolve(__dirname, 'stubs'),
+    mount: [['/js', path.join(clientPath, 'static/js')]],
+    open: false,
+    logLevel: 0
+  });
+  browser = await puppeteer.launch({
+    args: [
+      // Required for Docker version of Puppeteer
+      '--no-sandbox',
+      '--disable-setuid-sandbox',
+      // This will write shared memory files into /tmp instead of /dev/shm,
+      // because Docker’s default for /dev/shm is 64MB
+      '--disable-dev-shm-usage'
+      // dumpio: true
+    ]
+  });
+  global.Worker = createPseudoWorker(await newPageContext(browser));
+  page = await newPageContext(browser);
+  await page.setViewport({ width: 300, height: 150 });
+  const testLangs = testedLangs();
+  const langAndChallenges = await Promise.all(
+    testLangs.map(lang => getChallenges(lang))
+  );
+  const meta = {};
+  for (const { lang, challenges } of langAndChallenges) {
+    meta[lang] = {};
+    for (const challenge of challenges) {
+      const dashedBlockName = dasherize(challenge.block);
+      if (!meta[dashedBlockName]) {
+        meta[lang][dashedBlockName] = (await getMetaForBlock(
+          dashedBlockName
+        )).challengeOrder;
+      }
+    }
+  }
+  return {
+    meta,
+    langAndChallenges
+  };
+}
 
-async function runTests() {
+// cleanup calls some async functions, but it's the last thing that happens, so
+// no need to await anything.
+function cleanup() {
+  if (browser) {
+    browser.close();
+  }
+  liveServer.shutdown();
+  spinner.stop();
+}
+
+function runTests({ langAndChallenges, meta }) {
   process.on('unhandledRejection', err => {
-    spinner.stop();
     throw new Error(`unhandledRejection: ${err.name}, ${err.message}`);
   });
 
-  const testLangs = testedLangs();
+  if (process.env.npm_config_superblock && process.env.npm_config_block) {
+    throw new Error(`Please do not use both a block and superblock as input.`);
+  }
 
-  const challenges = await Promise.all(
-    testLangs.map(lang => getChallenges(lang))
-  );
+  // the next few statements will filter challenges based on command variables
+  if (process.env.npm_config_superblock) {
+    langAndChallenges[0].challenges = langAndChallenges[0].challenges.filter(
+      challenge => challenge.superBlock === process.env.npm_config_superblock
```

Comment:
Could this be looser?  A case insensitive regex would be easy enough to write and is a touch more user-friendly.

Reply:
Working on it.